cmd/lncli/commands.go

```diff
@@ -651,27 +656,30 @@ var closeChannelCommand = cli.Command{
 				"sat/byte that should be used when crafting " +
 				"the transaction",
 		},
+		cli.BoolFlag{
+			Name:  "all",
+			Usage: "close all existing active and inactive channels",
+		},
 	},
 	Action: actionDecorator(closeChannel),
 }
 
 func closeChannel(ctx *cli.Context) error {
-	ctxb := context.Background()
-	client, cleanUp := getClient(ctx)
-	defer cleanUp()
-
-	args := ctx.Args()
-	var (
-		txid string
-		err  error
-	)
-
-	// Show command help if no arguments provided
 	if ctx.NArg() == 0 && ctx.NumFlags() == 0 {
 		cli.ShowCommandHelp(ctx, "closechannel")
 		return nil
 	}
 
+	if ctx.Bool("all") {
+		// Make sure only the "all" flag was set.
+		if ctx.NArg() != 0 || ctx.NumFlags() != 1 {
```

Comment:
Can I suggest that we allow `—all` to be used in conjunction with `—force` so that the user doesn’t have to manually step through each inactive channel? Might even consider adding `—force-only` so that we can keep active channels!

Reply:
Definitely! I think it'd be better to create another command `closeallchannels` to avoid making the `force` flag have different uses depending on the context. This would also allow us to introduce additional flags, like for closing inactive channels only.